### PR TITLE
feat: add ide-extension for i18next handling

### PR DESCRIPTION
**Pull request by NilsJacobsen**
> Let me know if I need to change something to merge this PR

I added the inlang [IDE extension](https://inlang.com/documentation/apps/ide-extension) to sealos. Helps with handling translation keys and translation linting.

## What did I add?
- `inlang.config.js` to configure where language translations are
- Would recommend adding it to the `.vscode` folder so other contributors can use it.

### Extract Messages (translations)
Extract Messages (translations) via the Inlang: Extract Message code action.

Extract Messages

### Inline annotations
You can see translations directly in your code. No back-and-forth looking into the translation files themselves. (code example from sealos)

image

Ootb [editor](https://inlang.com/documentation/apps/web-editor) & [CLI](https://inlang.com/documentation/apps/inlang-cli) if needed.

### CODE REVIEW

**Comment by NilsJacobsen** [comment]
Hey btw sealos has 16 erros (missingMessage/ referenceMessageMissing) in the translation.

image

This could negatively effect the user experience.

**Comment by NilsJacobsen** [comment]
I also sow that the key `Recharge Amount` is often used but not in the translations. Probably an old key.

image

**Comment by fanux** [comment]
@c121914yu inlang seems to be a good method for i18n, please follow up on this pull request.

**Comment by NilsJacobsen** [comment]
@c121914yu 

> How do I get it started?
Do I need to install other packages? or IDE extension.
Please give a complete example.

After merge, you just need to open vscode with the inlang ide-extension installed. It should automatically work.
Usage is documented here -> https://inlang.com/documentation/apps/ide-extension

> Because we are using the i18n ally plugin locally. I'm not sure if inlang can be replaced it.

i18n-ally can also be used as a helper with i18n, but inlang is going for the infrastructure approach, which caught me. What does it mean:
- Comes with i18n products that are just usable ootb with the config. ([ide-extension](https://inlang.com/documentation/apps/ide-extension), [cli](https://inlang.com/documentation/apps/inlang-cli), [editor](https://inlang.com/documentation/apps/web-editor)) 
- no lock-in, for those who want to use i18n-ally it is not a problem
- inlang is being built fast, while i18n-ally is basically unmaintained by localize https://github.com/lokalise/i18n-ally/discussions/896

In conclusion, inlang can replace it. Especially with the release of https://github.com/inlang/inlang/issues/960 and https://github.com/inlang/inlang/issues/724. But as I said there is no need to decide (no lock-in).

**Comment by c121914yu** [comment]
I have added the inlang.config.js file locally and have installed the inlang extension. But didn't seem to see much effect.
I see an sdk in the inlang documentation,  do I need to install it? 
Or I need to add a configuration in.vscode/settings.json?

**Comment by NilsJacobsen** [comment]
> I see an sdk in the inlang documentation, do I need to install it?

The SDK is only relevant if you didn't localize yet. 

> Or I need to add a configuration in.vscode/settings.json?

There shouldn't be anything to change here.

You can test if the config is valid with `npx @inlang/cli config validate`. -> https://inlang.com/documentation/apps/inlang-cli#config-validate
If that works you should see the inline annotations. Sometimes you need to reload the window. While doing that on the bottom left corner it should say `inlang extension activated`. After some seconds it should show the annotations and of course, you could use the extract function.

**Comment by NilsJacobsen** [comment]
inlang did a big update which requires an update on this PR.

-> [Whats new](https://inlang.com/blog/refactoring-inlang)

Should I change it?

**Comment by c121914yu** [comment]
> inlang did a big update which requires an update on this PR.
> 
> -> [Whats new](https://inlang.com/blog/refactoring-inlang)
> 
> Should I change it?



> inlang did a big update which requires an update on this PR.
> 
> -> [Whats new](https://inlang.com/blog/refactoring-inlang)
> 
> Should I change it?

Thank you for your PR. But we don't feel we have a difficult problem with i18n-Ally right now, and we will keep watching on inlang's development and evaluate it.

We will re-evaluate the i18n when we rebuild the architecture in Q4

**Comment by c121914yu** [comment]
> inlang did a big update which requires an update on this PR.
> 
> -> [Whats new](https://inlang.com/blog/refactoring-inlang)
> 
> Should I change it?

Thank your PR. We will re-evaluate the i18n when we rebuild the architecture in Q4.

**Comment by NilsJacobsen** [comment]
Hi @c121914yu just wanted to check-in if you talked already about i18n in Q4?
